### PR TITLE
refactor(gigs): make gigs great again 

### PR DESCRIPTION
**Pull request by humzakhan**
# Description & Technical Solution

- All transactions are signed in `gigs-dialog` component 
- Other contract/gigs components work with the state supplied by this component

# Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] Already rebased against main branch.

# Screenshots

Provide screenshots or videos of the changes made if any.

### CODE REVIEW

**Review by ChristopherPark3 — APPROVED** [review]
Looks great! 🤌

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,15 +1,10 @@
 # Description & Technical Solution
```
Comment (Cutaiar) [inline .github/pull_request_template.md:1]:
```suggestion
# Description
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
 
-Describe problems, if any, clearly and concisely.
-Summarize the impact to the system.
-Please also include relevant motivation and context.
-Please include a summary of the technical solution and how it solves the problem.
-
 # Checklist
 
 - [ ] I have commented my code, particularly in hard-to-understand areas.
 - [ ] Already rebased against main branch.
```
Comment (Cutaiar) [inline .github/pull_request_template.md:7]:
```suggestion
```

Comment (Cutaiar) [reply .github/pull_request_template.md:7]: Would you mind while you're at it?

File: src/app/gigs/gigs-container.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ export interface GigsContainerProps {
   isClient?: boolean;
```
Comment (Cutaiar) [inline src/app/gigs/gigs-container.tsx:15]: Remove the interface if unused

File: src/hooks/gigs/useGetContractForOffer.ts [hunk 4]
```diff
@@ -11,12 +11,14 @@ type useGetContractForOfferParams = {
 export const useGetContractForOffer = (data: useGetContractForOfferParams) => {
   const { currentOfferId, isOpen, activeTab } = data;
 
-  return useQuery({
+  const query = useQuery({
```
Comment (Cutaiar) [inline src/hooks/gigs/useGetContractForOffer.ts:14]: Don't extract to a const if not needed.

File: src/app/gigs/contract/contract-overview.tsx [hunk 5]
```diff
@@ -232,11 +157,33 @@ export const ContractOverview = ({
                 projectId={contract.id}
                 isClient={isClient}
                 milestoneId={milestone.id}
+                handleMilestoneFunding={() =>
```
Comment (Cutaiar) [inline src/app/gigs/contract/contract-overview.tsx:160]: Consider following standard event handler naming convention. [More info](https://jaketrent.com/post/naming-event-handlers-react/).
```suggestion
                onMilestoneFunded={() =>
```
or something similar.

File: src/app/gigs/gigs-dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,704 @@
...
+  activeTab,
+  setActiveTab,
+}) => {
+  return (
+    <div className='border border-1 border-solid border-gray-100 rounded-full h-11'>
+      <Button
+        variant={`outline`}
+        className={cn(
+          'rounded-full border-none outline-none gap-2 active:ouline-none focus:outline-none hover:bg-sorbet hover:text-white',
+          `${activeTab === ActiveTab.Chat && `bg-sorbet text-white`}`
```
Comment (Cutaiar) [inline src/app/gigs/gigs-dialog.tsx:81]: `cn` can accept falsy values so you can just do
```suggestion
          activeTab === ActiveTab.Chat && 'bg-sorbet text-white'
```

File: src/app/gigs/gigs-dialog.tsx [hunk 7]
```diff
@@ -0,0 +1,704 @@
...
+        onClick={() => setActiveTab(ActiveTab.Chat)}
+      >
+        <span>Chat</span>
+        <IconMessage size={15} />
+      </Button>
+      <Button
+        variant={`outline`}
+        className={cn(
+          'rounded-full border-none gap-2 active:ouline-none focus:outline-none hover:bg-sorbet hover:text-white',
+          `${activeTab === ActiveTab.Contract && `bg-sorbet text-white`}`
```
Comment (Cutaiar) [inline src/app/gigs/gigs-dialog.tsx:92]:
```suggestion
          activeTab === ActiveTab.Contract && 'bg-sorbet text-white'
```

File: src/app/gigs/gigs-dialog.tsx [hunk 8]
```diff
@@ -0,0 +1,704 @@
...
+  Chat,
+  Contract,
+}
+
+interface TabSelectorProps {
+  activeTab: ActiveTab;
+  setActiveTab: (tab: ActiveTab) => void;
+}
+
+const TabSelector: React.FC<TabSelectorProps> = ({
```
Comment (Cutaiar) [inline src/app/gigs/gigs-dialog.tsx:71]: Minor suggestion is to locate local components below main exported component. This way, `GigsDialog` is visible at the top of the file and is next to it's props

File: src/app/gigs/gigs-dialog.tsx [hunk 9]
```diff
@@ -0,0 +1,704 @@
...
+  }, [contractData]);
+
+  const handleApprove = async () => {
+    setIsLoading(true);
+    if (contractData) {
+      const response = await updateContractStatus(
+        contractData.id,
+        'NotStarted'
+      );
+      if (response && response.data) {
```
Comment (Cutaiar) [inline src/app/gigs/gigs-dialog.tsx:175]: response?.data

File: src/app/gigs/gigs-dialog.tsx
Comment (Cutaiar) [inline src/app/gigs/gigs-dialog.tsx:1]: This component is almost 1000 lines long. Low priority, but I would consider splitting it over multiple files.

**Review by Cutaiar — APPROVED** [review]
Approved with suggestions ✨

File: src/app/gigs/gigs-dialog.tsx
Comment (humzakhan) [reply src/app/gigs/gigs-dialog.tsx:1]: no time to do that right now. Feel free to refactor in the future.

**Pull request merged**